src/WebJobs.Extensions.DurableTask/LifeCycleNotificationHelper.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal class LifeCycleNotificationHelper
+    {
+        private readonly DurableTaskExtension config;
+        private readonly ExtensionConfigContext extensionConfigContext;
+        private readonly string eventGridKeyValue;
+        private static HttpClient httpClient = null;
+
+        public LifeCycleNotificationHelper(DurableTaskExtension config, ExtensionConfigContext extensionConfigContext)
+        {
+            this.config = config ?? throw new ArgumentNullException(nameof(config));
+            this.extensionConfigContext = extensionConfigContext ?? throw new ArgumentNullException(nameof(extensionConfigContext));
+
+            if (!string.IsNullOrEmpty(config.EventGridTopicEndpoint) && !string.IsNullOrEmpty(config.EventGridKeySettingName))
```

Comment:
I suggest we add a bit more validation here in case users make a configuration mistake.
* If `EventGridTopicEndpoint` is set but `EventGridKeySettingName` is not, then we should throw an exception to inform the developer that a key is required.
* Also, further down below, we should throw an exception if `this.eventGridKeyValue` is null since it probably means that the developer had an incorrect app setting name.

Reply:
Done for it. Also I add a test for this fix. Please check my error message is proper or not.

Reply:
Great! I will suggest some improvements for the error messages.